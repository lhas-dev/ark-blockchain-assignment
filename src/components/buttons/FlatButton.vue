<template>
  <button class="flat-button" v-if="component === 'button'">
    <slot></slot>
  </button>
  <router-link class="flat-button" v-else :to="to"><slot></slot></router-link>
</template>

<script>
export default {
  props: {
    component: {
      type: String,
      default: "link"
    },
    to: {
      type: String
    }
  }
};
</script>
