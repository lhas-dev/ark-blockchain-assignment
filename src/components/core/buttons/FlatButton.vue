<template>
  <button
    class="text-lg p-3 text-gray-500 rounded hover:bg-gray-200 hover:text-gray-600"
  >
    <slot></slot>
  </button>
</template>

<script>
export default {};
</script>

<style></style>
