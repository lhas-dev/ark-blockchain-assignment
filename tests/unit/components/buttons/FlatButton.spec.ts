import { shallowMount } from "@vue/test-utils";
import FlatButton from "@/components/buttons/FlatButton.vue";

describe("FlatButton.vue", () => {
  it("renders correctly", () => {
    const wrapper = shallowMount(FlatButton, {
      stubs: ["router-link", "router-view"]
    });
    expect(wrapper).toMatchSnapshot();
  });
});
