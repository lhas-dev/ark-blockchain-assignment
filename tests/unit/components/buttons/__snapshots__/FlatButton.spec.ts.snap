// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlatButton.vue renders correctly 1`] = `<router-link-stub class="flat-button"></router-link-stub>`;
